src/components/ReportArticle.jsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { withRouter, Redirect } from 'react-router-dom';
+import propTypes from 'prop-types';
+import { toast } from 'react-toastify';
+import { reportArticle, clearErrors } from '../actions/reportArticleAction';
+
+
+class ReportArticle extends React.Component {
+  /**
+   * Creates an instance of ReportArticle class.
+   * @param {*} {props} - various props supplied to the component
+   * @memberof ReportArticle
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      type: '',
+      comment: '',
+    };
+  }
+
+  componentDidMount() {
+    const { success } = this.props;
+    if (success) {
```

Comment:
This works? wow. I've never thought of it in this way. I just learnt something new.